### ISSUE TITLE
Adjust default container selected for scroll

### ISSUE THREAD

**Issue by achhina** [tridactyl/tridactyl#5036]
Hi, I was asked to file an issue regarding the scrollable container selected not being what was expected. I find that usually the right container is selected, but an example of the wrong one being selected is for [WCAG](https://www.w3.org/TR/WCAG21/#intro) page. I've attached a screen recording showing the TOC being selected as the scroll container on page load, but correctly scrolling after selecting the appropriate one.

https://github.com/user-attachments/assets/e72a23df-9b88-4413-9182-173df5878784

Interestingly, the arrow keys select the main container when we visit a sub-heading, but otherwise also select the TOC when loaded from the main page. I believe this is because the TOC is the active selection without providing a sub-heading, but otherwise is the main body container with a sub-heading and then follows the [focus order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)?

While scrolling the TOC because it's the active container on page load _seems_ to be part of the WCAG spec from what I could parse, I still believe what I usually want is what was outlined in this [comment](https://github.com/tridactyl/tridactyl/issues/1805#issuecomment-632934826) by @bovine3dom

> - we scroll the widest element on the page first (i.e. j/k scroll wide elements, h/l scroll tall elements)
> - once that element is scrolled we switch to the next widest element
> - if focus changes we scroll that element instead.

However, I'm not sure how difficult/expensive this would be to calculate while traversing the nodes.

**Comment by achhina** [comment]
Just to add another data point, Sourcegraph is another example where the largest container does not get selected on page load, i.e. [sourcegraph/tridactyl/tridactyl](https://sourcegraph.com/github.com/tridactyl/tridactyl) where the sidebar for the source files tree is selected.

**Comment by bovine3dom** [comment]
Thanks, I can confirm I see the wrong container being scrolled on both pages.

You are probably already aware but you can use `;;` hint mode to change which part of a page is scrolled.

**Comment by achhina** [comment]
Yup! I'm happy with the scrolling heuristics 95% of the time, this was just related to our last discussion and thought I'd add datapoints as I came across them incase it's helpful.
